### DISCUSSION TITLE
User mode provision script has no access to LIMA_CIDATA_* vars?

### DISCUSSION THREAD

**Discussion by frkd-dev** [lima-vm/lima#2050] [Q&A]
I'm trying to use [Lima's environment vars](https://github.com/lima-vm/lima/blob/master/pkg/cidata/cidata.TEMPLATE.d/lima.env) from a user mode provision script like this:

```yaml
provision:
- mode: user
  script: |
    #!/bin/sh
    ln -s $LIMA_CIDATA_HOSTHOME_MOUNTPOINT/.conan ~/.conan
```

Unfortunately, this creates a wrong link like the var wasn't set:
```
# ls -la ~
total 20
drwxr-xr-x    4 zzzzz    zzzzz         4096 Dec  4 15:01 .
drwxr-xr-x    3 root     root          4096 Dec  2 00:02 ..
-rw-------    1 zzzzz    zzzzz          234 Dec  4 11:58 .ash_history
lrwxrwxrwx    1 zzzzz    zzzzz            7 Dec  4 15:01 .conan -> /.conan
drwxr-xr-x    3 zzzzz    zzzzz         4096 Dec  4 11:54 .config
drwx------    2 zzzzz    zzzzz         4096 Dec  2 00:02 .ssh
```
It seems like Lima's environment vars are not accessible from the user script while from "system" mode script they are. Template variables like `{{.Name}}` aren't usable as well.

Is this a bug or intended behavior? If intended behavior, then how to get LIMA_* vars from user mode provision scripts?

**Comment by frkd-dev** [comment]
I figured out that the user's script environment doesn't inherit LIMA_* variables as `sudo` [command lacks](https://github.com/lima-vm/lima/blob/b453dedb757e5fc1e249f352d0812d008e66fa3f/pkg/cidata/cidata.TEMPLATE.d/boot.sh#L69) `-E` parameter which preserves variables for the executed script.

I tend to think this was forgotten as this hugely limits user mode scripts. I'll create a PR that adds this.

**Comment by jandubois** [comment]
> Template variables like `{{.Name}}` aren't usable as well.

Template variables are available in provisioning scripts since the 0.20.0 release (#2121).

`{{ .Name }}` will be available in the 0.21.0 release (#2179).
